### PR TITLE
PMM-5135 Fix query examples for MySQL 8.0.

### PR DESCRIPTION
**Pull request by JiriCtvrtka**

### CODE REVIEW

File: agents/mysql/perfschema/perfschema.go [hunk 1]
```diff
@@ -206,8 +208,38 @@ func (m *PerfSchema) runHistoryCacheRefresher(ctx context.Context) {
 	}
 }
 
+func getMySQLVersion(q *reform.Querier) (mysqlVersion float64, vendor string, err error) {
```
Comment (percona-csalguero) [inline agents/mysql/perfschema/perfschema.go:211]: Just a comment. 
Usually we don't hve named return parameters in our functions. Maybe it is a topic to discuss.

**Review by percona-csalguero — APPROVED** [review]

File: agents/mysql/perfschema/perfschema.go [hunk 2]
```diff
@@ -206,8 +208,38 @@ func (m *PerfSchema) runHistoryCacheRefresher(ctx context.Context) {
...
+	var name, ver string
+	err = q.QueryRow(`SHOW /* pmm-agent-tests:MySQLVersion */ GLOBAL VARIABLES WHERE Variable_name = 'version'`).Scan(&name, &ver)
```
[the remark this answers is not in the document]
Comment (JiriCtvrtka) [reply agents/mysql/perfschema/perfschema.go:213]: No, thanks. Will fix it

File: agents/mysql/perfschema/perfschema.go [hunk 3]
```diff
@@ -206,8 +220,57 @@ func (m *PerfSchema) runHistoryCacheRefresher(ctx context.Context) {
...
+	mysqlVersion, err = strconv.ParseFloat(ver, 64)
+	return
+}
+
 func (m *PerfSchema) refreshHistoryCache() error {
-	current, err := getHistory(m.q)
+	var mysqlVersion float64
+	var vendor string
+	mutexRW.RLock()
+	if v, ok := versions[m.agentID]; ok {
```
Comment (BupycHuk) [inline agents/mysql/perfschema/perfschema.go:245]: no need to keep it in `versions` map, you could create new field in `PerfSchema` which would contain version and vendor of DB Server

File: utils/version/mysql.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+	mysqlDBRegexp = regexp.MustCompile(`^\d+\.\d+`)
+)
+
+// ParseMySQLVersion return parsed version of MySQL.
+func ParseMySQLVersion(v, c string) (string, string) {
+	m := mysqlDBRegexp.FindString(v)
+	parts := strings.Split(m, ".")
+	var version string
+	switch len(parts) {
+	case 1: // major only
```
Comment (BupycHuk) [inline utils/version/mysql.go:34]: it's hardly possible to have any value except 2 since your regexp is strict.

File: utils/version/mysql.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+	"strings"
+)
+
+// regexps to extract version numbers from the `SHOW GLOBAL VARIABLES WHERE Variable_name = 'version'` output.
+var (
+	mysqlDBRegexp = regexp.MustCompile(`^\d+\.\d+`)
+)
+
+// ParseMySQLVersion return parsed version of MySQL.
+func ParseMySQLVersion(v, c string) (string, string) {
```
Comment (BupycHuk) [inline utils/version/mysql.go:29]: We already have this logic in `utils/tests/mysql.go`, so you could just copy it from there.

File: agents/mysql/perfschema/perfschema.go (hunk 3 above)
Comment (JiriCtvrtka) [reply agents/mysql/perfschema/perfschema.go]: Fixed.

File: utils/version/mysql.go (hunk 4 above)
Comment (JiriCtvrtka) [reply utils/version/mysql.go]: Fixed.

File: utils/version/mysql.go (hunk 5 above)
Comment (JiriCtvrtka) [reply utils/version/mysql.go]: Done.

**Comment by JiriCtvrtka** [comment]
@BupycHuk I did required changes. Merging it because today is last day. If you will want another changes I will create ticket.

**Pull request merged**